addons/crossword/test/EventTests.js

```diff
@@ -71,12 +71,38 @@ TestCase("[Crossword] Events tests", {
         assertFalse(this.presenter.isGradualShowAnswersActive);
     },
 
+    'test given Show Answers when gaps edited then user answers are properly saved': function () {
+        this.presenter.isWordNumbersCorrect = () => true;
+        this.buildCrossword();
+
+        this.presenter.$view.find('.cell_' + 5 + 'x' + 5 + ' input')
+            .val("A");
+        this.presenter.showAnswers();
+
+        const expected = "A";
+        const actual = this.presenter.userAnswers[5][5];
+
+        assertTrue(expected === actual);
+    },
+
+    'test given Hide Answers then user answers are properly brought back': function () {
```

Comment:
Tutaj zabrakło w formule given-when-then słówka when ale to nie jest super istotne.

Co do całości to czytelność wzrosła chyba o 100% po wydzieleniu kodu do funkcji , good job :D